src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/home/SharedHomeFormula.kt

```diff
@@ -37,6 +47,31 @@ internal class SharedHomeFormula(
             it.safeExecute("sudo mv $jiraHome/logos $localSharedHome")
             ubuntu.install(it, listOf("nfs-kernel-server"))
             it.execute("sudo echo '$localSharedHome $localSubnet(rw,sync,no_subtree_check,no_root_squash)' | sudo tee -a /etc/exports")
+
+            if (s3StorageBucketName != null && storeInS3.isNotEmpty()) {
+                AwsCli(awsCliVersion).ensureAwsCli(it)
+                if (storeInS3.contains(ATTACHMENTS)) {
+                    // Copy the attachment data from NFS onto S3
+                    // Use xargs to split the work across 30 concurrent jobs.  This is much faster than a single job.
+                    it.safeExecute(
+                        cmd = "export AWS_RETRY_MODE=standard; export AWS_MAX_ATTEMPTS=10; cd $localSharedHome/data && ( find attachments -mindepth 1 -maxdepth 1 -type d -print0 | xargs -n1 -0 -P30 -I {} aws s3 cp --recursive --only-show-errors {}/ s3://$s3StorageBucketName/{}/ )",
```

Comment:
Not necessarly actionable in this PR - just FYI / future work note for other reviewers.

This is hard to unit test. I created suggestion for us to consider: https://ecosystem.atlassian.net/browse/JPERF-1122

Reply:
Yeah it's a bit dense that line.  We could likely achieve something similar if we could write jvm code to run on the remote as in your suggestion.  That would make it much more testable too.

Reply:
This should be improved. Preference effort/quality starting with lowest
- extracted into a separated method
- injected as an object of a new class and unit tested
- @pczuj's suggestion https://ecosystem.atlassian.net/browse/JPERF-1122